framework/src/wifi_manager/wifi_profile.c

```diff
@@ -0,0 +1,238 @@
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <dirent.h>
+#include <sys/types.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <wifi_manager/wifi_manager.h>
+#ifdef CONFIG_WIFI_PROFILE_SECURESTORAGE
+#include <mbedtls/see_api.h>
+#endif
+#include "wifi_common.h"
+#include "wifi_profile.h"
+
+//#define WIFI_PROFILE_USE_ETC
+#define WIFI_PROFILE_PATH "/mnt/"
+#define WIFI_PROFILE_FILENAME "wifi.conf"
+
+#define DELIMITER "\t"
+#define DELI_LEN 1
+
+#ifdef CONFIG_WIFI_PROFILE_SECURESTORAGE
+#define WIFI_PROFILE_SS_INDEX 1
+#endif
+
+#define ENCODE_STRING(buf, size, data, pos)					\
+	do {													\
+		snprintf(buf + pos, size, "%s" DELIMITER, data);	\
+		pos += (strlen(data) + DELI_LEN);					\
+	} while (0)
+
+#define ENCODE_INTEGER(buf, size, data, pos)				\
+	do {													\
+		char conv[12];										\
+		snprintf(buf + pos, size, "%d" DELIMITER, data);	\
+		sprintf(conv, "%d", data);							\
+		pos += (strlen(conv) + DELI_LEN);					\
+	} while (0)
+
+#define DECODE_STRING(buf, data, pos)				\
+	do {											\
+		sscanf(buf + pos, "%s", data);				\
+		pos += ((int)(strlen(data)) + DELI_LEN);	\
+	} while (0)
+
+#define DECODE_INTEGER(buf, data, pos)			\
+	do {										\
+		char conv[12];							\
+		int tmp;								\
+		sscanf(buf + pos, "%d", &tmp);			\
+		sprintf(conv, "%d", tmp);				\
+		data = tmp;								\
+		pos += (strlen(conv) + DELI_LEN);		\
+	} while (0)
+
+/*
+ * Internal Functions
+ */
+
+int _wifi_profile_serialize(char *buf, uint32_t buf_size, wifi_manager_ap_config_s *config)
```

Comment:
This looks internal API. If it is, let's add ```static```. Same for all of ```_xxx``` function in this file

Reply:
Ok I will fix